src/main/java/at/ac/tuwien/kr/alpha/common/atoms/BasicAtom.java

```diff
@@ -44,14 +44,14 @@ public BasicAtom(Predicate predicate, List<Term> terms) {
 		this(predicate, terms, false);
 	}
 
-	public BasicAtom(BasicAtom clone) {
-		this(clone, clone.negated);
+	public BasicAtom(Literal clone) {
+		this(clone, clone.isNegated());
 	}
 
-	public BasicAtom(BasicAtom clone, boolean negated) {
+	public BasicAtom(Atom clone, boolean negated) {
```

Comment:
This looks very dangerous, because you can then call `new BasicAtom(new BuiltinAtom(...))` and you get a very weird result.